src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -146,36 +157,42 @@ public AutoMLExperiment SetPipeline(SweepableEstimatorPipeline pipeline)
 
         public AutoMLExperiment SetEvaluateMetric(BinaryClassificationMetric metric, string labelColumn = "label", string predictedColumn = "Predicted")
         {
-            _settings.EvaluateMetric = new BinaryMetricSettings()
+            var metricManager = new BinaryMetricManager()
             {
                 Metric = metric,
                 PredictedColumn = predictedColumn,
                 LabelColumn = labelColumn,
             };
+            _serviceCollection.AddSingleton<IMetricManager>(metricManager);
+            this.SetIsMaximizeMetric(metricManager.IsMaximize);
 
             return this;
         }
 
         public AutoMLExperiment SetEvaluateMetric(MulticlassClassificationMetric metric, string labelColumn = "label", string predictedColumn = "Predicted")
         {
-            _settings.EvaluateMetric = new MultiClassMetricSettings()
+            var metricManager = new MultiClassMetricManager()
             {
                 Metric = metric,
                 PredictedColumn = predictedColumn,
                 LabelColumn = labelColumn,
             };
+            _serviceCollection.AddSingleton<IMetricManager>(metricManager);
+            this.SetIsMaximizeMetric(metricManager.IsMaximize);
 
             return this;
         }
 
         public AutoMLExperiment SetEvaluateMetric(RegressionMetric metric, string labelColumn = "label", string scoreColumn = "Score")
         {
-            _settings.EvaluateMetric = new RegressionMetricSettings()
+            var metricManager = new RegressionMetricManager()
```

Comment:
Since you are no longer assigning values to `_settings.EvaluateMetric` or `_settings.DatasetSettings`, does the settings class still need those members?

Reply:
Nevermind, I see further down you did remove them. Resolving this comment.